example/include/AverageCurrent.h

```diff
@@ -0,0 +1,20 @@
+// SPDX-FileCopyrightText: Deutsches Elektronen-Synchrotron DESY, MSK, ChimeraTK Project <chimeratk-support@desy.de>
+// SPDX-License-Identifier: LGPL-3.0-or-later
+#pragma once
+
+#include <ChimeraTK/ApplicationCore/ApplicationCore.h>
+
+namespace ctk = ChimeraTK;
+
+//! [Class Definition]
+class AverageCurrent : public ctk::ApplicationModule {
+  using ctk::ApplicationModule::ApplicationModule;
+
+  // Take the heaterCurrent as output by the Controller as an input
+  ctk::ScalarPushInput<float> current{this, "../Controller/heatingCurrent", "mA", "Actuator output of the controller"};
```

Comment:
Just to show that this is also possible.

Now I am rethinking this a bit, also in light of your comment about the ModuleGroup. Maybe we should:

- Remove the ModuleGroup around the SetpointRamp and make the SetpointRamp optional itself.
- Place both the Controller and the AverageCurrent into one ModuleGroup.
- The relative path then makes perfectly sense, because otherwise it would depend on the name of the surrounding ModuleGroup
- Downsides: Have to introduce the concept of ModuleGroups quite early, and it makes the process variable hierarchy more complicated (e.g. "/Oven/Controller/heatingCurrent" if "Oven" is the name of the ModuleGroup).

What do you think?

Reply:
Let's keep it as it is at the moment. The other one can be more confusion. Why would an oven have a module group 'Oven'. And I usually have module groups to have the whole group optional, or multiple instances of it. Otherwise I might not even group it (in such a simple case).

Sure, we want to show as many concepts as possible. But it has to be intuitive from the example why we do it.
When you learn, you start by copying. If we introduce lots of 'wild' hierarchy  just because we can, people will start writing overly complicated servers.